### PR TITLE
Various Fixes and enhancements

### PR DESCRIPTION
**Pull request by prem12321kumar**
- added padding in feelsLike
- Czech date formatting
- Changed margin of weather condition icon to position it for better appearing
- pressing the "Enter" key on the keyboard in any input field(location, api, proxy) performs the same action as clicking the corresponding button.
say we entered a location, to save it we click on "enter" button, now we can save by pressing enter key from Keyboard too.
- userText placeholder text removal when the user starts editing and restore placeholder if the user leaves it empty after clicking, --> now works for all languages
- fixed: When refreshed or new tab is opened, "Double Tap to Edit" shows briefly before loading
- changed 'enter' to 'save' and 'search'
- translated 'save' and 'search' in all lang
- reordered language.js and script.js in index.html
- added translation function for greeting with fallback to english
- Translated greeting in all languages

Edit:
-Added Translation for menuCloseButton
-added translation for 'close' in all languages
with fallback to english if translation unavailable
-and some weather variable named

Edit2:
-Corrected some Czech transaltion with @ceskyDJ 
-Bengali Numerals
alongwith option to add any other language numeral too

edit3:
-placeholder inputtext color
-Digital clock: display AM/PM when using 12 hr format
    localised numerals and texts
    date formatting
    am/pm color
   
-Immediately date is updated when clock is analog

edit4:
-Replace decimal point with a comma for Czech
-Added a space before unit in Feelslike temperature in Czech
-changed "Enable/Disable" to "Show/Hide" for shortcuts and AI tools (in English only)

edt5:
-Fallback to English if any elements translation is missing (Earlier when any elements' translation is missing in all lang it shows eng but if it defined in even 1 lang then for other lang it was showing undefined, now with all the fallbacks it will show in English)
-locale am/pm

-Fixed: Weather Display issue when weatherCondition is long

-no leading zeros in hour for 12 hr format in digital clock, except in English and Italian

-Digital Clock: Hours in darker color and blinking colon

### CODE REVIEW

**Comment by prem12321kumar** [comment]
@XengShi 😅

**Review by ZiClaud — APPROVED** [review]

**Comment by prem12321kumar** [comment]
@ceskyDJ

**Comment by ceskyDJ** [comment]
Czech has more day phases ("ráno", "dopoledne", "poledne", "odpoledne", "večer", "noc"), but I think this could be sufficient.

**Comment by prem12321kumar** [comment]
@ceskyDJ 'Zavřiť' for close in menuCloseText
correct?

**Comment by ceskyDJ** [comment]
Almost, the correct one is "Zavřít“ (diacritical "i").

**Comment by prem12321kumar** [comment]
> Almost, the correct one is "Zavřít“ (diacritical "i" and normal "t").

corrected, thanks

**Review by ceskyDJ — APPROVED** [review]

**Comment by ceskyDJ** [comment]
> corrected, thanks

You're welcome.

File: languages.js [hunk 1]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+
+        // Weather
+        "humidityLevel": "Vlhkost",
+        "feelsLike": "Pocitová teplota",
+        "location": "Země",
+        // End of Weather
+
+        // New Tab Item
+        "conditionText": "Dobrý den! Jak se máte?",
+        "enterBtn": "Hledat",
```
Comment (ceskyDJ) [inline languages.js:370]:
```suggestion
        "enterBtn": "Vyhledat",
```

Comment (ceskyDJ) [reply languages.js:370]: It's better than "Hledat" in this context.

**Comment by prem12321kumar** [comment]
@itz-rj-here Translated Bengali Numerals

**Comment by prem12321kumar** [comment]
I think all major elements' text are now translated

beside obvious ones, digital clock (and alerts from proxy and reset setting) left

File: languages.js [hunk 2]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "shortcutsInfoText": "Vyberte, které zkratky se mají zobrazit",
+        "adaptiveIconText": "Adaptivní tvary ikon",
+        "adaptiveIconInfoText": "Ikony zkratek budou vždy kulaté",
+        "ai_tools_button": "AI nástroje",
+        "enable_ai_tools": "Povolí/zakáže zkratky AI nástrojů",
+        // Digital Clock
+        "digitalclocktittle": "Digitální hodiny",
+        "digitalclockinfo": "Povolí/zakáže digitální hodiny",
+        "timeformattittle": "Použít 12hodinový formát",
+        "timeformatinfo": "Použije 12hodinový formát času",
```
Comment (ceskyDJ) [inline languages.js:329]:
```suggestion
        "timeformatinfo": "Použije se 12hodinový formát času",
```

File: languages.js [hunk 3]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
-    // End of Body and New Tab Items
-},
-        
+        // Menu Items
+        "feedback": "Zpětná vazba",
+        "resetsettings": "Resetovat nastavení",
+        menuCloseText: 'Zavřít',
+        // Shortcuts
+        "shortcutsText": "Zkratky",
+        "enableShortcutsText": "Povolí/zakáže zkratky",
```
Comment (ceskyDJ) [inline languages.js:317]:
```suggestion
        "enableShortcutsText": "Zapne/vypne zkratky",
```

File: languages.js [hunk 4]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "editShortcutsText": "Upravit zkratky",
+        "editShortcutsList": "Upravit zkratky",
+        "shortcutsInfoText": "Vyberte, které zkratky se mají zobrazit",
+        "adaptiveIconText": "Adaptivní tvary ikon",
+        "adaptiveIconInfoText": "Ikony zkratek budou vždy kulaté",
+        "ai_tools_button": "AI nástroje",
+        "enable_ai_tools": "Povolí/zakáže zkratky AI nástrojů",
```
Comment (ceskyDJ) [inline languages.js:324]:
```suggestion
        "enable_ai_tools": "Zapne/vypne zkratky AI nástrojů",
```

Comment (ceskyDJ) [reply languages.js:324]: Maybe "Show" is better for this?

File: languages.js (hunk 3 above)
Comment (ceskyDJ) [reply languages.js:317]: Maybe "Show" is better for this?

File: languages.js [hunk 5]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        // Digital Clock
+        "digitalclocktittle": "Digitální hodiny",
+        "digitalclockinfo": "Povolí/zakáže digitální hodiny",
```
Comment (ceskyDJ) [inline languages.js:327]:
```suggestion
        "digitalclockinfo": "Zapne/vypne digitální hodiny",
```

File: languages.js (hunk 3 above)
Comment (ceskyDJ) [reply languages.js:317]: This translation is for the current state; the last message is just a proposal for change.

File: languages.js (hunk 4 above)
Comment (ceskyDJ) [reply languages.js:324]: This translation is for the current state; the last message is just a proposal for change.

File: languages.js (hunk 5 above)
Comment (ceskyDJ) [reply languages.js:327]: I'd change it to "Switches to digital clock", but here it's pretty ok right now. My translation string change refers to the current state.

File: languages.js [hunk 6]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "timeformattittle": "Použít 12hodinový formát",
+        "timeformatinfo": "Použije 12hodinový formát času",
+        // Misc
+        "fahrenheitCelsiusCheckbox": "Přepnout na stupně Fahrenheita",
+        "fahrenheitCelsiusText": "Změny se projeví po obnovení stránky",
+        "search_suggestions_button": "Návrhy vyhledávání",
```
Comment (ceskyDJ) [inline languages.js:333]:
```suggestion
        "search_suggestions_button": "Návrhy ve vyhledávání",
```

File: languages.js [hunk 7]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "search_suggestions_text": "Povolí/zakáže návrhy vyhledávání",
```
Comment (ceskyDJ) [inline languages.js:334]:
```suggestion
        "search_suggestions_text": "Zapne/vypne návrhy vyhledávání",
```

File: languages.js [hunk 8]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        // Proxy
+        "useproxytitletext": "Obcházení proxy",
+        "useproxyText": "Pokud nefungují návrhy vyhledávání",
```
Comment (ceskyDJ) [inline languages.js:337]:
```suggestion
        "useproxyText": "Pokud nefungují návrhy ve vyhledávání",
```

File: languages.js [hunk 9]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "ProxyText": "Proxy pro obcházení CORS",
+        "ProxySubtext": "Přidejte vlastní proxy pro obcházení CORS",
```
Comment (ceskyDJ) [inline languages.js:339]:
```suggestion
        "ProxySubtext": "Nastavte si vlastní proxy pro obcházení CORS",
```

File: languages.js [hunk 10]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "HostproxyButton": "Provozování vlastní proxy",
+        "saveproxy": "Uložit",
+        // Location
+        "UserLocText": "Zadejte svou polohu",
+        "UserLocSubtext": "Pokud není poloha počasí správná",
```
Comment (ceskyDJ) [inline languages.js:344]:
```suggestion
        "UserLocSubtext": "Pokud není správná poloha počasí",
```

File: languages.js [hunk 11]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "location": "Země",
+        // End of Weather
+
+        // New Tab Item
+        "conditionText": "Dobrý den! Jak se máte?",
+        "enterBtn": "Vyhledat",
+        "searchPlaceholder": "Zadejte hledaný výraz...",
+        "searchWithHint": "Vyhledávat prostřednictvím",
+        "ai_tools": "AI nástroje",
+        "userText": "Klikněte sem pro úpravu",
```
Comment (ceskyDJ) [inline languages.js:374]:
```suggestion
        "userText": "Upravíte kliknutím",
```

**Comment by ceskyDJ** [comment]
I have found some strings with bad Czech translations (resp. too machine-like, unnatural). I don't know if I overlooked some strings before or if you rewrote them via machine translation, but it'd be good if the proposed fix is accepted.

File: languages.js (hunk 11 above)
Comment (prem12321kumar) [reply languages.js:374]: #103 MTL

**Comment by prem12321kumar** [comment]
> I have found some strings with bad Czech translations (resp. too machine-like, unnatural). I don't know if I overlooked some strings before or if you rewrote them via machine translation, but it'd be good if the proposed fix is accepted.
> 
Nah, I didn't rewrite anything. I just double-checked to confirm, lol, if I pasted old translation.

> Maybe an edition for clearance. The Czech language uses a very different way of building sentences, and some things are casually said less or more differently than in English. I try to translate it like it is written in Czech from the start, so it's friendly to Czech speakers and sounds natural (like Czech, not machine-translated English).
>
Exactly, there's no need to follow the English translation word-for-word—just aim for what sounds perfect and natural.

**Comment by ceskyDJ** [comment]
> Nah, I didn't rewrite anything. I just double-checked to confirm, lol, if I pasted the old translation.

Could be. I have an experience that translations need to be done in many iterations to be of good quality :D. So, maybe it all falls just on my head. I still think there were some strings that changed from my last edit (maybe before creating this pull request). I just wanted to let you (and others) know it's not good to replace already done translations with machine-generated ones. It's better to let me know (e.g., tag me), and I'll update it if it's necessary.

**Comment by prem12321kumar** [comment]
Also I think for shortcuts and AI tools, "Show/Hide" would be ideal since it determines whether they are visible or not.
For the digital clock and search suggestions, "Enable/Disable" is better as it affects their functionality or availability, not just their visibility.
I'll change them tomorrow morning

**Comment by ceskyDJ** [comment]
Ok, I'll update Czech strings after.

**Review by itz-rj-here — APPROVED** [review]
Looks good to go.

**Comment by itz-rj-here** [comment]
> @itz-rj-here Translated Bengali Numerals

Great work.

**Comment by prem12321kumar** [comment]
@ceskyDJ is short date format in digital clock correct?
image

**Review by itz-rj-here — APPROVED** [review]

**Review by itz-rj-here — APPROVED** [review]
Wow good work.

**Comment by prem12321kumar** [comment]
#97 fixed

**Comment by prem12321kumar** [comment]
> Wow good work.

Thanks 😌

**Comment by prem12321kumar** [comment]
@XengShi  I’m all done for now. Please review my changes, and feel free to suggest any modifications. I'm happy to provide explanations for any part of the updates.

**Comment by XengShi** [comment]
> @XengShi  I’m all done for now. Please review my changes, and feel free to suggest any modifications. I'm happy to provide explanations for any part of the updates.

Okay, Currently I am kinda busy, outside my home, can't access my pc...

File: languages.js [hunk 12]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        // Shortcuts
+        "shortcutsText": "Zkratky",
+        "enableShortcutsText": "Zapne/vypne zkratky",
+        "editShortcutsText": "Upravit zkratky",
+        "editShortcutsList": "Upravit zkratky",
+        "shortcutsInfoText": "Vyberte, které zkratky se mají zobrazit",
+        "adaptiveIconText": "Adaptivní tvary ikon",
+        "adaptiveIconInfoText": "Ikony zkratek budou vždy kulaté",
+        "ai_tools_button": "AI nástroje",
+        "enable_ai_tools": "Zapne/vypne zkratky AI nástrojů",
```
Comment (ceskyDJ) [inline languages.js:324]:
```suggestion
        "enable_ai_tools": "Zobrazí/skryje zkratky AI nástrojů",
```

File: languages.js [hunk 13]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
-    // End of Body and New Tab Items
-},
-        
+        // Menu Items
+        "feedback": "Zpětná vazba",
+        "resetsettings": "Resetovat nastavení",
+        menuCloseText: 'Zavřít',
+        // Shortcuts
+        "shortcutsText": "Zkratky",
+        "enableShortcutsText": "Zapne/vypne zkratky",
```
Comment (ceskyDJ) [inline languages.js:317]:
```suggestion
        "enableShortcutsText": "Zobrazí/skryje zkratky",
```

File: languages.js [hunk 14]
```diff
@@ -262,233 +284,264 @@ const translations = {
...
+        "location": "Země",
+        // End of Weather
+
+        // New Tab Item
+        "conditionText": "Dobrý den! Jak se máte?",
+        "enterBtn": "Vyhledat",
+        "searchPlaceholder": "Zadejte hledaný výraz...",
+        "searchWithHint": "Vyhledávat prostřednictvím",
+        "ai_tools": "AI nástroje",
+        "userText": "Upravíte kliknutím",
```
Comment (ceskyDJ) [inline languages.js:374]:
```suggestion
        "userText": "Upravíte po kliknutí",
```
Hah, this is even better :D.

**Comment by prem12321kumar** [comment]
> Okay, Currently I am kinda busy, outside my home, can't access my pc...

No worries at all! Take your time.

**Comment by ceskyDJ** [comment]
> @ceskyDJ is short date format in digital clock correct?

I don't know what you want to say through the short-date format. Is is `%name_of_day_in_week%, %day_in_month%.`? In this case, it's correct.

PM for 12-hour format is correct (odp.).

**Comment by ceskyDJ** [comment]
> I don't know what you want to say through the short-date format. Is is `%name_of_day_in_week%, %day_in_month%.`? In this case, it's correct.

I looked at the English variant to see what it should print. It's ok ;).

**Comment by prem12321kumar** [comment]
> PM for 12-hour format is correct (odp.) with one exception – we don't use leading zeros in place of hours in time (we use it just for minutes), so these are correct: 13:00, 1:00, 1:30.

yeah, leading zeros dont generally used in other lang too. ill try to change it.
From search: In 12-hour format, a leading zero is usually context-dependent in English and Italian but not commonly used in most other languages. In the 24-hour format, the leading zero is almost universally used.

**Comment by prem12321kumar** [comment]
> > PM for 12-hour format is correct (odp.) with one exception – we don't use leading zeros in place of hours in time (we use it just for minutes), so these are correct: 13:00, 1:00, 1:30.
> 
> yeah, leading zeros dont generally used in other lang too. ill try to change it. From search: In 12-hour format, a leading zero is usually context-dependent in English and Italian but not commonly used in most other languages. In the 24-hour format, the leading zero is almost universally used.

Done
you can check it at https://prem12321kumar.github.io/materialYouNewTab/

**Review by itz-rj-here — CHANGES_REQUESTED** [review]
Can you change the twitter shortcut icon to x.com icon?

**Comment by itz-rj-here** [comment]
As the twitter.com is no longer being used, it redirects to the x.com web.

**Comment by prem12321kumar** [comment]
> Can you change the twitter shortcut icon to x.com icon?

I noticed it but I don't know about svg path,
Just googled and found it, I'll try in sometime if I can do.

**Comment by XengShi** [comment]
> Can you change the twitter shortcut icon to x.com icon?

That little bird 🐦 is prettier than 𝕏, so I think we should still keep using that. 🙃🙃🙃

**Comment by itz-rj-here** [comment]
> > Can you change the twitter shortcut icon to x.com icon?
> 
> I noticed it but I don't know about svg path, Just googled and found it, I'll try in sometime if I can do.

I see. looks like @XengShi didn't liked the idea. :) anyways then keep it as it is.

**Comment by prem12321kumar** [comment]
> > Can you change the twitter shortcut icon to x.com icon?
> 
> That little bird 🐦 is prettier than 𝕏, so I think we should still keep using that. 🙃🙃🙃


Ok lol, I also like the old icon and name, 
Will just change the redirect link
no name or icon change

**Comment by XengShi** [comment]
> 
> > > Can you change the twitter shortcut icon to x.com icon?
> > 
> > I noticed it but I don't know about svg path, Just googled and found it, I'll try in sometime if I can do.
> 
> I see. looks like @XengShi didn't liked the idea. :) anyways then keep it as it is.

If you want it, simply use "𝕏" (copy & paste) and replace it with twitter svg.

**Comment by XengShi** [comment]
You've done a lot of work, improving translation accuracy and other aspects, but unfortunately, I'm still facing the same issue we encountered earlier.
The weather elements are still getting stuck on 'Seems' and 'Earth.'


https://github.com/user-attachments/assets/c6353346-cdcd-48e7-b5a1-65345536cac6

@itz-rj-here Can you test this repo on your machine

**Comment by itz-rj-here** [comment]
> You've done a lot of work, improving translation accuracy and other aspects, but unfortunately, I'm still facing the same issue we encountered earlier. The weather elements are still getting stuck on 'Seems' and 'Earth.'
> 
>  2024-10-26.11-34-16.mp4 
> @itz-rj-here Can you test this repo on your machine

Just a min let me download it and check it.

**Comment by prem12321kumar** [comment]
> You've done a lot of work, improving translation accuracy and other aspects, but unfortunately, I'm still facing the same issue we encountered earlier. The weather elements are still getting stuck on 'Seems' and 'Earth.'
> 
>  2024-10-26.11-34-16.mp4 
> @itz-rj-here Can you test this repo on your machine

that's odd, it's working fine for me on Chrome. I'll look into the code.

Can you please confirm if :main has the same issue or not.

https://xengshi.github.io/materialYouNewTab/
and https://prem12321kumar.github.io/materialYouNewTab/
both have correct display but when refreshed, the same display like yours. I thought at first it maybe because of some issue with virtual deployment.

**Comment by itz-rj-here** [comment]
https://screenrec.com/share/7YtjxoNDeF

**Comment by itz-rj-here** [comment]
> You've done a lot of work, improving translation accuracy and other aspects, but unfortunately, I'm still facing the same issue we encountered earlier. The weather elements are still getting stuck on 'Seems' and 'Earth.'
> 
>  2024-10-26.11-34-16.mp4 
> @itz-rj-here Can you test this repo on your machine

It currently works fine for me. Just the weather icon sometimes doesn't load cuz of the loading timeout.

**Comment by itz-rj-here** [comment]
@prem12321kumar can i know why you replaced the weather icon little bit of downwards?

**Comment by itz-rj-here** [comment]
W8 i found something weird. let me screen record and show ya all.

**Comment by itz-rj-here** [comment]
https://screenrec.com/share/G08aQb2hru

When I use a vpn and connect to uk ip, my weatherapi runs properly. But in my own ip the weatherapi sometimes doesn't able to give correct data and stays blank.

**Comment by prem12321kumar** [comment]
> * position it for better appearing

It was appearing a bit off to me, one sided.
It is highlighted especially in clear night one

https://github.com/user-attachments/assets/2a866143-b7dc-4659-8258-9acb1004c8ee


Is it not fine?

**Comment by prem12321kumar** [comment]
> https://screenrec.com/share/G08aQb2hru
> 
> When I use a vpn and connect to uk ip, my weatherapi runs properly. But in my own ip the weatherapi sometimes doesn't able to give correct data and stays blank.

huh, here no problem with display of Humidity, Feels, Earth 
but weather icon not loading but loaded with vpn... can't understand reason
Try with a different API?

**Comment by ceskyDJ** [comment]
> Try with a different API?

Maybe download images and serve them locally? Is it possible?

**Comment by itz-rj-here** [comment]
> > https://screenrec.com/share/G08aQb2hru
> > 
> > When I use a vpn and connect to uk ip, my weatherapi runs properly. But in my own ip the weatherapi sometimes doesn't able to give correct data and stays blank.
> 
> huh, here no problem with display of Humidity, Feels, Earth 
> but weather icon not loading but loaded with vpn... can't understand reason
> Try with a different API?

It happens cause the ip of the weatherapi takes time to connect with my ip.

**Comment by prem12321kumar** [comment]
> It happens cause the ip of the weatherapi takes time to connect with my ip.

if so then maybe other data also shouldn't have loaded

can you try deleting cookies and site data of weatherapi.com
image
then open new tab and press Ctrl + Shift + R

**Comment by prem12321kumar** [comment]
@XengShi what is the error in the console?

**Comment by prem12321kumar** [comment]
> @XengShi what is the error in the console?

Do you face the same issue in XengShi:main/v2.2, or is it working fine?

**Comment by itz-rj-here** [comment]
> > @XengShi what is the error in the console?
> 
> Do you face the same issue in XengShi:main/v2.2, or is it working fine?

https://screenrec.com/share/uPQ6t0fdAH
https://screenrec.com/share/fYztmTpOMo

**Comment by itz-rj-here** [comment]
I faced some issues here.

**Comment by itz-rj-here** [comment]
Also noticed that you have replaced the icon at little bit of downwards, which doesn't look good. Also ur weather icon has issue. Not on the original one as i noticed.

**Comment by itz-rj-here** [comment]
I think you should look on this and try to add some other weather api's too.
https://github.com/PranshulGG/WeatherMaster/tree/v1.7.4?tab=readme-ov-file#-weather-providers

**Comment by XengShi** [comment]
@prem12321kumar Today I rechecked it again, and everything works! not getting any error.

**Comment by XengShi** [comment]
> > @XengShi what is the error in the console?
> 
> Do you face the same issue in XengShi:main/v2.2, or is it working fine?

[Windows 10, Chrome]
On the first reload or opening a new tab, it shows the location and 'Feels like' temperature for a few milliseconds, but then it goes back to displaying 'Earth' and 'Feels,' even though the temperature is shown correctly (in the pill-shaped widget).


Screenshot 2024-10-28 043052
image

**Comment by prem12321kumar** [comment]
> > > @XengShi what is the error in the console?
> > 
> > 
> > Do you face the same issue in XengShi:main/v2.2, or is it working fine?
> 
> [Windows 10, Chrome] On the first reload or opening a new tab, it shows the location and 'Feels like' temperature for a few milliseconds, but then it goes back to displaying 'Earth' and 'Feels,' even though the temperature is shown correctly (in the pill-shaped widget).
> 
> Screenshot 2024-10-28 043052 image

#113 
Please try resetting shortcuts 
IMG_20241028_102337

**Pull request merged**